### PR TITLE
contentful: notify users when their code uses removed rich text fields

### PR DESCRIPTION
**Pull request by axe312ger**
This should help people that upgrade and miss the migration steps in the changelog

### CODE REVIEW

File: packages/gatsby-source-contentful/src/normalize.js [hunk 1]
```diff
@@ -483,9 +484,19 @@ exports.createNodesForContentType = ({
               )
             })
 
+          const deprecationReason = `This field got removed due to performance reasons. Please use the new "raw" and "references" fields. See: https://www.gatsbyjs.com/plugins/gatsby-source-contentful/#contentful-rich-text`
```
Comment (stefanjudis) [inline packages/gatsby-source-contentful/src/normalize.js:487]:
```suggestion
          const deprecationReason = `This field was removed in 
gatsby-source-contentful v4. Please use the new "raw" and "references" fields. See: https://www.gatsbyjs.com/plugins/gatsby-source-contentful/#contentful-rich-text`
```

Works for me. If you want you can take a tiny adjustment, but no hard feelings he. :)

**Review by stefanjudis — APPROVED** [review]
🚢

File: packages/gatsby-source-contentful/src/normalize.js [hunk 2]
```diff
@@ -483,9 +484,20 @@ exports.createNodesForContentType = ({
               )
             })
 
+          const deprecationReason = `This field was removed in 
+gatsby-source-contentful v4. Please use the new "raw" and "references" fields. See: https://www.gatsbyjs.com/plugins/gatsby-source-contentful/#contentful-rich-text`
+
```
Comment (axe312ger) [inline packages/gatsby-source-contentful/src/normalize.js:489]: 🤦 pretty sure thats the wrong position for this 😅

**Comment by axe312ger** [comment]
Hmm, this won't work like that. As Rich Text fields are no nodes anymore, I can't extend the schema to add deprecation reasons.

To solve this we'd need create the GraphQL schema via the schema customisation API which is way to much effort to achieve a deprecation message.

Does anybody else have an idea how to notify users that their query is using old fields and where to look for migration instructions?

**Comment by KyleAMathews** [comment]
Perhaps add the field back and then throw and error if someone tries to query it? That's a very strong depreciation message haha.

**Comment by axe312ger** [comment]
The old fields will now return the deprecations string.

I can't use real deprecationReason as these fields are generated automagically and there is no RichTextNodeType (anymore)


We can properly add deprecation messages in as soon the plugin creates the GraphQL schema based on the content model, not the content itself.

**Comment by axe312ger** [comment]
Sorry, this is not trivial. We will let users know via documentation.

**Pull request closed without merging**